doc/using/configuration.chapter.md

```diff
@@ -191,6 +191,13 @@ list-id: configuration-variable-list
 source: ../config-options.json
 ```
 
+## Calling Nixpkgs from the Module System {#sec-invoke-nixpkgs-by-module}
+
+For [module system](#module-system) application authors, Nixpkgs provides a reusable module for the purpose of invoking Nixpkgs. Most users should not have to be aware of this. The documentation for its options is meant to be re-exposed in the application's documentation and is therefore not repeated here.
```

Comment:
```suggestion
For [module system](#module-system) application authors, Nixpkgs provides a reusable module for the purpose of configuring Nixpkgs. Other users should not have to be aware of this. Therefore, the documentation of its options is not rendered here. Module system authors should re-expose it in their own documentation.
```

Reply:
The main clarification is `configuring` instead of `invoking`. The rest just tries to amend a bit more to the reader's ductus.

Reply:
"Module system application authors" is precisely as intended.
I'd characterize the audience in roughly descending guesstimated size

- module system users
- module authors
- module system application authors
- module system contributors (although larger if you consider `types.nix` beyond `submodule`, which we technically should)

Reply:
If useful, for later: [diagram](https://play.d2lang.com/?script=jFI7isMwEO11ineBvYCLBbO1qyUHkO1JENgzQiOFhOC7B_lDLEihTozeG72PWu8nN9gcWYOXAToZ00S4KIVtAKwKG3gKKmyA5QP7E47B9SlKDbpNObpqIH5-SznEg4ykoIfPPpRwlfBVzIl6PDoct6Tr2XGSpHCzD3KnmTiW7k8rCpe3YMdk8xbHUcxi_p8aaS7S20Y4hVvvfed2-Utbx1U17JyqNg7B2V_RPrHtJ1LzDgAA__8%3D&)

Reply:
> * module system contributors (although larger if you consider `types.nix` beyond `submodule`, which we technically should)

This one is irrelevant. It's rare for the module system itself to change.

As for your diagram, it's not clear to me whether the System box refers to the module system or part of a module system application.

It's unfortunate the "system" is already part of the "module system" name. I think a system of modules would be a nice description of a module system application...

Perhaps a more interesting classification of domains is
 - **core modules**: set of modules that everything in a module system application depends on. Examples in NixOS:
    - everything in top-level
    - systemd perhaps
 - **integration modules**: modules that simplify the integration of a large number of things
    - services
    - programs
    - image formats perhaps (not sure about the status of these tbh. loading these as modules isn't always perfect)
 - **user modules**
    - a "user"'s configuration
    - any modules they wrote for their configuration
 - **module system**: mostly irrelevant; it's the air we breathe
    - `_module` options
    - a subset of `lib`

Reply:
New [diagram](https://play.d2lang.com/?script=jFPBatxADL37Kx65tD0kue8hEPbUQ6EQ-gGztmwLbMloNGmWkn8vM7Z3bTaE3MZjPT299zTP0zRwHbJlB_yrgAV0QDxHpxHaYtQmDRQr4Fc54U8km6uBMv4BE1lUqYD3a9lRxY1PyfUr1c8p-_rlQtw_7cchqbWhCHqbsshIaNU-HGYDXUnr9S_FcmZJmiJ4nExfaSTxvfpNi53KzkKTQu7C4lpVwFGNltKYUT_FqbNi-Pb6mKLruL35bdrycn5J7CWAbbfVqmti5Ju44H1w5MTP3rN0YEFYfq7hhmv6aGgiaSIudn8w6Q3ljivyOA3cnuE9Zf0XtLYIGPKuQtJ4Iss3Zai4cO3l39AE3KVIdvct7hm9pzP-mnoJO3-y5fxa7pIFv0hZvfys8RXGF9zs-w1qmtshdJ1RN2tkKbLz8tCbz5JrldrICSSvbCp5j_CdHroHjKHuWQiPqAdNDR5BXv8otO_Vslkvc0j3T9i9UpJwys-x-h8AAP__&) :smile: :handshake:

Reply:
Lovely. What's a suite though?

The addition of profiles is nice. User modules tend to be like profiles, but I don't know if everyone agrees on what a profile is... And we should get rid of the term because of `nix profile`.

Maybe _suite_ is a nice word for that, so I'm very curious. If you could briefly define profile and suite, that would be sweet.

Reply:
I think profile is a `haumea.load`-able. Essentially an almost-`json`-serializable. User Data. Definitions, not declarations.

For bigger projects "suites" may be handy as "profile"-aggregators. For example: the teacher suite may have the student suite + some management profiles.

Reply:
Ohh it has tooltips! :facepalm: 

Too bad _profile_ is already rather overloaded in the Nix world, so I'd be inclined to call them suites even if they only contribute to one option value.